lib/FrontendTool/FrontendTool.cpp

```diff
@@ -1275,23 +1296,9 @@ static bool performCompileStepsPostSILGen(CompilerInstance &Instance,
     return processCommandLineAndRunImmediately(
         Invocation, Instance, std::move(SM), MSF, observer, ReturnValue);
 
-  // FIXME: We shouldn't need to use the global context here, but
-  // something is persisting across calls to performIRGeneration.
-  auto &LLVMContext = getGlobalLLVMContext();
-  std::unique_ptr<llvm::Module> IRModule;
-  llvm::GlobalVariable *HashGlobal;
-  if (MSF.is<SourceFile*>()) {
-    IRModule = performIRGeneration(IRGenOpts,
-                                   *MSF.get<SourceFile*>(),
-                                   std::move(SM),
-                                   opts.InputsAndOutputs.getSingleOutputFilename(), LLVMContext,
-                                   0, &HashGlobal);
-  } else {
-    IRModule = performIRGeneration(IRGenOpts, MSF.get<ModuleDecl*>(),
-                                   std::move(SM),
-                                   opts.InputsAndOutputs.getSingleOutputFilename(), LLVMContext,
-                                   &HashGlobal);
-  }
+  std::pair<std::unique_ptr<llvm::Module>, llvm::GlobalVariable *>
+      IRModuleAndHashGlobal =
```

Comment:
I would probably still use an out parameter rather than a `std::pair` here to not deal with `std::get`, but I can see why you'd prefer this. But please unpack it right after the return.

Reply:
Changed to use 2 out parameters.